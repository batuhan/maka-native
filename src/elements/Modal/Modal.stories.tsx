import React from "react";
import { View, Text } from "react-native";
import { storiesOf } from "@storybook/react-native";

import useModal from "../../lib/hooks/useModal";

import Modal from "./Modal";

const ModalContainer = () => {
  const { isShowing, toggle } = useModal();
  return (
    <View>
      <Text onPress={toggle}>Show popup</Text>
      <Modal show={isShowing}>
        <Text onPress={toggle}>Close popup</Text>
      </Modal>
    </View>
  );
};
storiesOf("Modal", module).add("Modal", () => (
  <View>
    <ModalContainer />
  </View>
));
