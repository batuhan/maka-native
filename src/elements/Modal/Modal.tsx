import React, { ReactNode } from "react";
import { Modal } from "react-native";
import styled from "styled-components";

import Container from "../../elements/Container/Container";

export interface Props {
  show: boolean;
  children: ReactNode;
}
const ModalComponent = ({ show, children }: Props) => {
  return (
    <Container>
      <Modal animationType="slide" visible={show} transparent={false}>
        {children}
      </Modal>
    </Container>
  );
};

export default ModalComponent;
