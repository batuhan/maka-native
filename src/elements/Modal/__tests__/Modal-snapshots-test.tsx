import React from "react";
import renderer from "react-test-renderer";
import { Text } from "react-native";

import Modal from "../Modal";

describe("Modal", () => {
  it("renders correctly", () => {
    const instance = renderer.create(
      <Modal show={false}>
        <Text>Modal Component</Text>
      </Modal>
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
