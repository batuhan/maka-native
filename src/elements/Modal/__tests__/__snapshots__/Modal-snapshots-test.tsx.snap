// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<View
  className="sc-bdVaJa kbrIhl"
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={false}
    visible={false}
  >
    <Text>
      Modal Component
    </Text>
  </Modal>
</View>
`;
